src/components/scene/background.js

```diff
@@ -0,0 +1,9 @@
+/* global THREE */
+var register = require('../../core/component').registerComponent;
+
+module.exports.Component = register('background', {
+  schema: {color: {type: 'color', default: 'black'}},
+  update: function () {
+    this.el.object3D.background = new THREE.Color(this.data.color);
```

Comment:
Why not `this.el.renderer.setClearColor(this.data.color)`?

Reply:
Because @mrdoob says so: `three.js r87 will bring a optimized handling of scene.background and should work nicely in VR.` in https://github.com/aframevr/aframe/issues/2908 😄 There's a ]WebGLBackground](https://github.com/mrdoob/three.js/blob/dev/src/renderers/webgl/WebGLBackground.js#L26-L108) thant handles `setClearColor` and a bunch of other background types like textures or cubemaps

Reply:
@dmarcos I was just following the logic of THREE API that `this.el.renderer.setClearColor(this.data.color)` would enable use of alpha and for supported textures `background` component could have `src`.

- `WebGLRenderer` imports [WebGLBackground#L13](https://github.com/mrdoob/three.js/blob/dev/src/renderers/WebGLRenderer.js#L13)
```
import { WebGLBackground } from './webgl/WebGLBackground.js';
```

- which it is assigned to [WebGLRenderer.background#L285](https://github.com/mrdoob/three.js/blob/dev/src/renderers/WebGLRenderer.js#L285)
```
background = new WebGLBackground( _this, state, geometries, _premultipliedAlpha );
```

-  WebGLRenderer.setClearColor calls [WebGLBackground#L450](https://github.com/mrdoob/three.js/blob/dev/src/renderers/WebGLRenderer.js#L450)
```
WebGLRenderer.setClearColor = function () {
  background.setClearColor.apply( background, arguments );
};
```

- [WebGLBackground.js#L17](https://github.com/mrdoob/three.js/blob/dev/src/renderers/webgl/WebGLBackground.js#L17) creates THREE.Color anyway
```
var clearColor = new Color( 0x000000 );
```
so passing `this.data.color` will result call to [clearColor.set#L116](https://github.com/mrdoob/three.js/blob/dev/src/renderers/webgl/WebGLBackground.js#L116)
```
WebGLBackground.clearColor.set( color );
```
and setting it as scene.background will copy color over to that [clearColor#L105](https://github.com/mrdoob/three.js/blob/dev/src/renderers/webgl/WebGLBackground.js#L105)  
```
state.buffers.color.setClear( color.r, color.g, color.b, alpha, premultipliedAlpha );
```

Reply:
@mrdoob yes https://github.com/aframevr/aframe/blob/fd843a1e842db0eb3b2c3ffb33ffcb23896e2402/src/core/scene/a-scene.js#L477-L481

Reply:
Ohm... I wonder if that's a good idea. In three.js we set alpha to false by default for performance/memory reasons.

In the case of a-frame, the only argument I can think of is that users may get confused because results may be different inside/outside VR. If they set alpha to true, the background of their 3d world will be the color of the html behind the element. When presenting in VR, only the canvas element is "composited", so the background would be black instead.

Reply:
For AR, you often want to be able to pass through to the background with alpha.  It is a good question whether VR should use the same setting for consistency, or use something else for better performance.  I would tend toward allowing the latter...

Reply:
> For AR, you often want to be able to pass through to the background with alpha. 

I think that is currently a hack though. I think it would be better to pass the video feed as a texture to your 3d world. The GPU is going to have to do the compositing one way or another. Better to do the compositing in your code than relaying on the browser's layout/compositing to deal with your scene.

Reply:
Discussion on this point has gone back and forth, for example there are privacy concerns - in some contexts it is better / safer for the camera data to be opaque to the application, even though that may be limiting for some use cases.

Reply:
(There are also passthrough AR devices such as Meta 2 that need to be able to show through, not rendering a texture)

Reply:
I see. Sounds like this issue has been settled before then.

Reply:
I think as we see the WebXR API evolve, which makes distinctions between VR, camera AR, and passthrough AR, we may have the opportunity to reintroduce some of these optimizations when we know they are safe